### ISSUE TITLE
Deprecate the --log-format flag in merge-logs command

### ISSUE THREAD

**Issue by arjunmahishi** [cockroachdb/cockroach#131496]
**Describe the problem**

The `--log-format` flag in the `debug merge-logs` and `debug zip upload` commands are confusing. 

After digging into this further, it was found that:
- The debug zip always generates the logs in `crdb-v1` format. Take a look at [this part](https://github.com/cockroachdb/cockroach/blob/e8d2bde5d66317bc12a8d232ac68e471e2616aa8/pkg/cli/zip_per_node.go#L496-L498) of the code.
    - `crdb-v1` is a legacy format now. `v21.1` onwards, `crdb-v2` is the main format ([ref](https://github.com/cockroachdb/cockroach/blob/c41d4a7e6669484597730d392f94c730f808cc1d/pkg/util/log/format_crdb_v2.go#L115)).
    - Also, the conversion to `crdb-v1` is lossy as it has a limited set of fields compared to newer formats.
- [This PR comment](https://github.com/cockroachdb/cockroach/pull/66629#pullrequestreview-687584912) touches on the above point.
- This has something to do with the fact that the stderr logs are in a different custom format. the `crdb-v1` parser seems to be capable of handling the custom format. But not the other parsers?

**Expected behavior**

- Ideally, the format specified in the log files header should be used to determine the format to parse the logs in.
- Look into the parsing of stderr files in v1 and v2 and determining if v2 is missing some special casing to handle them. 

**Slack**: https://cockroachlabs.slack.com/archives/C07KT4F00EA/p1727276143662199

**Comment by kevin-v-ngo** [comment]
Before we prioritize this issue, let's make sure we sync up with the TSE team since changing from the Debug zip crdb-v1 default format can be a breaking changing to existing troubleshooting workflows.

FYI @mattcrdb @smcvey @kevinkokomani

**Comment by arjunmahishi** [comment]
Makes sense. I also just want to point out this nuance: The logs are stored in two places. 
1. The disks of the nodes (in the format configured by the user)
2. The debug zip (which is always in `crdb-v1` format)

We have the capability to translate logs from one format to another. The reason this issue is proposing to move away from `crdb-v1` while creating the `debug.zip` is because this format is legacy. It supports very limited fields. So, converting from `crdb-v1` to a more recent version (with more fields) is lossy. If the `debug.zip` stores a newer format, and the troubleshooting workflows can only understand `crdb-v1`, we can always translate them at the last mile. For example, the `merge-logs` command could potentially support a `--output-format` flag. This would also allow us to make the troubleshooting workflow format agnostic.
